be/src/olap/rowset/segment_v2/parsed_page.h

```diff
@@ -56,6 +57,10 @@ struct ParsedPage {
         RETURN_IF_ERROR(encoding->create_page_decoder(data_slice, opts, &page->data_decoder));
         RETURN_IF_ERROR(page->data_decoder->init());
 
+        if (auto dict_decoder = dynamic_cast<BinaryDictPageDecoder*>(page->data_decoder)) {
```

Comment:
I am not sure we could use dynamic_cast to check if the decoder is dict page decoder.
I think we'd better check EncodingInfo first and then cast the data decoder to dict decoder.

}